R/convenience-functions.R

```diff
@@ -204,6 +205,10 @@ transform_forecasts <- function(data,
 
 log_shift <- function(x, offset = 0, base = exp(1)) {
 
+  assert_numeric(x, min.len = 1)
+  assert_number(offset)
+  assert_number(base)
```

Comment:
This will error if base is complex

```r
try(checkmate::assert_number(1 + (0+2i)))
#> Error in try(checkmate::assert_number(1 + (0 + (0+2i)))) : 
#>   Assertion on '1 + (0 + (0+2i))' failed: Must be of type 'number', not 'complex'.
Created on 2024-03-07 with [reprex v2.1.0](https://reprex.tidyverse.org/)
```

Do you want to do the following instead?

```r
assert(
check_number(base),
is.complex(base)
)
```

Reply:
- I think it might be good to check it here in case `base` is doing some weird coercion if it's neither a number nor complex (I've not checked but won't be surprised). Alternatively, if you decide to not check here, could you leave a quick comment there explaining that the check is remitted to `base::log()`? This will help a future reviewer not comment on it if they're wondering.

- Aside: you could inherit the documentation of `base::log()` from base to keep it aligned with whatever `base` expects and however they choose to document it. However, that is an issue for a different PR.

Reply:
@jamesmbaazam are you fine with this?

Reply:
That's fine with me. Could you update the documentation of `base` to say it only expects a number?

Reply:
Treating this as a "yes" :) 
<img width="192" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/144e11ae-3332-4ff8-9f2e-4448ed6ceec2">